src/mapcraftercore/mc/chunk.cpp

```diff
@@ -273,8 +268,16 @@ bool Chunk::readNBT(mc::BlockStateRegistry& block_registry, const char* data, si
 			continue;
 		}
 
-		std::copy(block_light.payload.begin(), block_light.payload.end(), section.block_light);
-		std::copy(sky_light.payload.begin(), sky_light.payload.end(), section.sky_light);
+               if(section_tag.hasArray<nbt::TagByteArray>("BlockLight", 2048))
```

Comment:
Please use tabs for indentation here as well. And look how I placed the curly brackets in the remaining code, also note the space between `if` and the condition.